### PR TITLE
add DRY-RUN-MODE for new addon upgrade to cc cable

### PR DESCRIPTION
**Pull request by m1zzx2**
#### What this PR does

add a DRY-RUN-MODE env. If FILTER_BY_ADDON_TYPE is enabled and DRY_RUN_MODE is ledacy, the result of dry-run will be legacy addon secret

#### why we need it
When the management cluster is upgraded from legacy to cc capable, according to this [doc](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.6/vmware-tanzu-kubernetes-grid-16/GUID-upgrade-tkg-index.html), the new addon needs to be installed by dry-run generated the secret of the legacy addon. 
The cli of ccbase does not have ability to generate legacy addon secret.
So this pr add a ability to generate addon legacy secrets under ccbase.

### CODE REVIEW

File: tkg/tkgctl/create_cluster.go [hunk 1]
```diff
@@ -166,6 +166,11 @@ func (t *tkgctl) processManagementClusterInputFile(ir *InitRegionOptions) (bool,
 			t.overrideManagementClusterOptionsWithLatestEnvironmentConfigurationValues(ir)
 		}
 	}
+
+	log.Info("GenerateOnly :%v is enable flag : %v env: %v", t.tkgClient.IsFeatureActivated(constants.FeatureFlagGenerateLegacyAddonManifest), os.Getenv("FILTER_BY_ADDON_TYPE"))
+	if ir.GenerateOnly && t.tkgClient.IsFeatureActivated(constants.FeatureFlagGenerateLegacyAddonManifest) && os.Getenv("FILTER_BY_ADDON_TYPE") != "" {
+		isInputFileClusterClassBased = false
```
Comment (DanielXiao) [inline tkg/tkgctl/create_cluster.go:172]: I don't think var isInputFileClusterClassBased control render legacy or cc cluster.

File: tkg/tkgctl/create_cluster.go [hunk 2]
```diff
@@ -263,6 +262,7 @@ func (t *tkgctl) getCreateClusterOptions(name string, cc *CreateClusterOptions,
 		IsWindowsWorkloadCluster:     cc.IsWindowsWorkloadCluster,
 		ClusterConfigFile:            cc.ClusterConfigFile,
 		IsInputFileClusterClassBased: isInputFileClusterClassBased,
+		IsGenerateOnly:               cc.GenerateOnly,
```
[the remark this answers is not in the document]
Comment (m1zzx2) [reply tkg/tkgctl/create_cluster.go:265]: Do you mean to remove this variable?
1.  there need a place to finger out if it is dry-run
2. After the upgrade, these secrets need to be installed on the workload cluster

File: tkg/client/init.go [hunk 3]
```diff
@@ -811,7 +811,7 @@ func (c *TkgClient) BuildRegionalClusterConfiguration(options *InitRegionOptions
 	if options.IsInputFileClusterClassBased {
 		bytes, err = getContentFromInputFile(options.ClusterConfigFile)
 	} else {
-		bytes, err = c.getClusterConfigurationBytes(&clusterConfigOptions, clusterConfigOptions.ProviderRepositorySource.InfrastructureProvider, true, false)
+		bytes, err = c.getClusterConfigurationBytes(&clusterConfigOptions, clusterConfigOptions.ProviderRepositorySource.InfrastructureProvider, true, false, options.GenerateOnly)
```
Comment (DanielXiao) [inline tkg/client/init.go:814]: I prefer to change only the management cluster.
```
		dryRunMode, _ := c.readerwriterConfigClient.TKGConfigReaderWriter().Get("DRY_RUN_MODE")
		filterByAddonType, _ := c.readerwriterConfigClient.TKGConfigReaderWriter().Get("FILTER_BY_ADDON_TYPE")
		if options.GenerateOnly && dryRunMode == "legacy" && filterByAddonType != "" {
			bytes, err = c.getClusterConfiguration(&clusterConfigOptions, true, clusterConfigOptions.ProviderRepositorySource.InfrastructureProvider, false)
		} else {
			bytes, err = c.getClusterConfigurationBytes(&clusterConfigOptions, clusterConfigOptions.ProviderRepositorySource.InfrastructureProvider, true, false)
		}
```

File: tkg/client/cluster.go [hunk 4]
```diff
@@ -199,12 +199,17 @@ func (c *TkgClient) CreateCluster(options *CreateClusterOptions, waitForCluster
...
 // getClusterConfigurationBytes returns cluster configuration by taking into consideration of legacy vs clusterclass based cluster creation
-func (c *TkgClient) getClusterConfigurationBytes(options *ClusterConfigOptions, infraProviderName string, isManagementCluster, isWindowsWorkloadCluster bool) ([]byte, error) {
+func (c *TkgClient) getClusterConfigurationBytes(options *ClusterConfigOptions, infraProviderName string, isManagementCluster, isWindowsWorkloadCluster, isGenerateOnly bool) ([]byte, error) {
 	deployClusterClassBasedCluster, err := c.ShouldDeployClusterClassBasedCluster(isManagementCluster)
 	if err != nil {
 		return nil, err
 	}
 
+	// if DRY_RUN_MODE is cc_capable, generate addon manifest under cc capable mode, will not enable ccbase
+	if isGenerateOnly && os.Getenv("DRY_RUN_MODE") == "cc_capable" && os.Getenv("FILTER_BY_ADDON_TYPE") != "" {
```
Comment (DanielXiao) [inline tkg/client/cluster.go:209]: Do not use hardcode string DRY_RUN_MODE". Use const like ConfigVariableDefaultBomFile

**Review by DanielXiao — APPROVED** [review]
You can merge it after fix the last comment.

**Pull request merged**